test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,478 @@
+            <Capability Type="Shell"></Capability>
+        </Resource>
+    </Resources>
+    <Quotas MaxIdleTimeoutms="2147483646" MaxConcurrentUsers="2147483646" IdleTimeoutms="6200000" MaxProcessesPerShell="2147483646"
+        MaxMemoryPerShellMB="2147483646" MaxConcurrentCommandsPerShell="2147483646" MaxShells="2147483646" MaxShellsPerUser="2147483646"></Quotas>
+</PlugInConfiguration>
+"@
+            Set-Content -Path $testdrive\plugin.xml -Value $fileXml
+            try {
+                $plugin = New-Item -Path WSMan:\localhost\Plugin -File $testdrive\plugin.xml -Name TestPlugin2
+                Test-Plugin -Plugin $plugin
+            }
+            finally {
+                Remove-Item WSMan:\localhost\Plugin\TestPlugin2\
+                "WSMan:\localhost\Plugin\TestPlugin2\" | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a plugin resource" {
+            try {
+                $resource = New-Item -Path WSMan:\localhost\Plugin\TestPlugin\Resources\ `
+                    -ResourceUri http://foo -Capability shell
+                $resource.PSPath | Should Exist
+                $properties = Get-ChildItem $resource.PSPath
+                ($properties | Where-Object { $_.Name -eq "ResourceUri" }).Value | Should Be "http://foo/"
+                ($properties | Where-Object { $_.Name -eq "Capability" })[0].Value | Should Be "shell"
+            }
+            finally {
+                Remove-Item $resource.PSPath
+                $resource.PSPath | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for an initialization parameter" {
+            try {
+                $parameter = New-Item -Path WSMan:\localhost\Plugin\TestPlugin\InitializationParameters `
+                    -ParamName foo -ParamValue bar
+                $parameterObj = Get-Item $parameter.PSPath
+                $parameterObj.Name | Should Be "foo"
+                $parameterObj.Value | Should Be "bar"
+            }
+            finally {
+                Remove-Item $parameter.PSPath
+                $parameter.PSPath | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a security resource" {
+            try {
+                $sddl = "O:NSG:BAD:P(A;;GA;;;BA)"
+                $resource = Get-ChildItem -Path WSMan:\localhost\Plugin\TestPlugin\Resources\ | Select-Object -First 1
+                # remove existing security resource since the folder name is just a hash of the resource uri
+                Get-ChildItem "$($resource.PSPath)\Security" | Remove-Item
+                $security = New-Item "$($resource.PSPath)\Security" -SDDL $sddl -Force
+                $security.PSPath | Should Exist
+                $securityObj = Get-Item $security.PSPath
+                (Get-ChildItem $securityObj.PSPath | Where-Object { $_.Name -eq 'sddl' }).Value | Should Be $sddl
+            }
+            finally {
+                Remove-Item $security.PSPath
```

Comment:
Does Remove-Item on a non-existant path throw an error?

Reply:
I'm testing Remove-Item and also trying to cleanup.  I think it may be better since @iSazonov gave similar feedback and have some redundancy where I do the `remove-item` test within the `try` and still do cleanup.